src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -177,6 +191,43 @@ void Instance::ReportOperationalStateListChange()
 void Instance::ReportPhaseListChange()
 {
     MatterReportingAttributeChangeCallback(ConcreteAttributePath(mEndpointId, mClusterId, Attributes::PhaseList::Id));
+    UpdateCountdownTimeFromClusterLogic();
+}
+
+void Instance::UpdateCountdownTime(bool fromDelegate)
+{
+    app::DataModel::Nullable<uint32_t> newCountdownTime = mDelegate->GetCountdownTime();
+    auto now = System::SystemClock().GetMonotonicTimestamp();
+
+    bool markDirty = false;
+
+    if (fromDelegate)
+    {
+        // Updates from delegate are reduce-reported to every 10s max (choice of this implementation), in addition
+        // to default change-from-null, change-from-zero and increment policy.
+        auto predicate = [](const decltype(mCountdownTime)::SufficientChangePredicateCandidate &candidate) -> bool {
+            if (candidate.lastDirtyValue.IsNull() || candidate.newValue.IsNull())
+            {
+                return false;
+            }
+
+            uint32_t lastDirtyValue = candidate.lastDirtyValue.Value();
+            uint32_t newValue = candidate.newValue.Value();
+            uint32_t kNumSecondsDeltaToReport = 10;
+            return (newValue < lastDirtyValue) && ((lastDirtyValue - newValue) > kNumSecondsDeltaToReport);
+        };
+        markDirty = (mCountdownTime.SetValue(newCountdownTime, now, predicate) == AttributeDirtyState::kMustReport);
+    }
+    else
+    {
+        auto predicate = [](const decltype(mCountdownTime)::SufficientChangePredicateCandidate &) -> bool { return true; };
```

Comment:
So all cluster-logic updates, including for random derived commands, force reporting?

Reply:
They do not force reporting, unless there was a change. Since your request in the QuieterReporting PR was to only take the `true` return of predicate to mean "report for sure on change, with no additional checks other than change". In other words, this is considered, if called from internal logic, as the internal knowledge of the cluster.

Reply:
No, here's what I mean.  Let's say a random derived-cluster command gets executed.  Since in that case `Instance::InvokeCommand` calls `UpdateCountdownTimeFromClusterLogic()`, we will report the CountdownTime if it's changed at all from the last-reported value, no matter how much time has passed since the last report, and completely ignoring the actual `kMarkDirtyOnIncrement | kMarkDirtyOnChangeToFromZero` policy.  In particular, we will report even if all that happened is the time ticked down normally.

What the spec says is:

> Changes to this attribute merely due to the normal passage of time with no other dynamic change of device state SHALL NOT be reported.

If a derived cluster command does not involve any change of device state (e.g. it's just a data request), then we would be violating that part of the spec.

I actually kind of have similar questions about the call in `OnOperationalErrorDetected` (spec doesn't seem to say that should report, if it didn't change the CountdownTime to 0 or null?), `OnOperationCompletionDetected` (again, if the phase/state did not change in the process should this really report?) and `ReportPhaseListChange` (why would this on its own trigger a report?).

And a similar concern in `HandleStartState`.  If we are currently running, why would receiving that command (which is a no-op in that situation) trigger an immediate report of the CountdownTime (yes, assuming that's changed since the last report)?

Reply:
Sure, I could report less. You are basically assuming ill intent from the cluster developer here. What would you recommend we do? I can take out quite a few of these, but I needed to at least have the edges of operational states report current time if changed (even if by one), as that is from important edges. In practice they won't change every second.

Reply:
I'm not assuming ill intent at all. I am assuming the cluster developer is trying their best to get correct behavior for their derived cluster.  But they can't control reporting because we are overriding it here.

And again: even within just this cluster, the reporting is not correct in `HandleStartState` already.

> but I needed to at least have the edges of operational states report current time if changed (even if by one),

That gets handled by the fact that a change in the operational state value triggers reporting, no?

Reply:
Left them out from commands. Fixed.